packages/block-editor/src/components/inserter/block-list.js

```diff
@@ -206,12 +206,12 @@ function InserterBlockList( {
 		);
 	}, [ filteredItems, rootChildBlocks ] );
 
-	const suggestedItems = useMemo( () => {
+	const suggestedItems = () => {
```

Comment:
I'm not sure what's the expect impact this change will have. That useMemo removal won't change the result of `suggestedItems` it will just ensure the instance don't change if it doesn't need to change.

Reply:
As explained in the issue linked above all the lists have to empty at some point for the `!hasItems` to pass.  On load `suggestedItems` is computed based on the `items` prop which is a list of all the registered blocks. The `items` list doesn't change and therefore this is only computed once. It therefore never returns empty and the `hasItems` boolean is always false.

Reply:
I would have to look at the code tomorrow, but it's possible that we may want to provide the `filteredItems` as opposed to the `items`.  But i'm not 100% on how that list is constructed.

Reply:
Ok I think I understand a bit better. It seems like the `suggesteddItems` need to be empty if there's no filter right?

In that case I'll just add this to the function inside use memo (if ! filter return []) and add the filter as a dependency. 

It seems like this was the behavior previously and it was missed in the refactor.

Reply:
And actually now that I think about it, that's why the test is passing. We are passing the empty array on component load which doesn't account for filtering the list... which would expose the issue with re-rendering.

Reply:
Alternatively,  we could possible change:
https://github.com/WordPress/gutenberg/blob/37995eb7c0d73299a248292b75a1863597fd1ce4/packages/block-editor/src/components/inserter/block-list.js#L326
to be: ` const hasItems = ! isEmpty( filteretedItems )` and leave the useMemo as it is because in reality that is definitely a better approach moving forward. The `hasItems` is pretty opaque at the moment.

Reply:
This works for me if it fixes your issue.